src/pages/home/report/ReportActionsView.js

```diff
@@ -174,53 +176,79 @@ class ReportActionsView extends React.Component {
-                // Otherwise counter will be shown on scrolling up from the bottom even if user have read those messages
-                if (this.state.isFloatingMessageCounterVisible) {
-                    this.updateMessageCounterCount(!isReportFullyVisible);
-                }
 
-                // Show new floating message counter when there is a new message
-                this.toggleFloatingMessageCounter();
-            }
-
-            // When the last action changes, record the max action
-            // This will make the NEW marker line go away if you receive comments in the same chat you're looking at
-            if (isReportFullyVisible) {
-                Report.readNewestAction(this.props.report.reportID);
+                // If the current user sends a new message in the chat we clear the new marker since they have "read" the report
+                this.setState({newMarkerSequenceNumber: 0});
+            } else if (isReportFullyVisible) {
+                // NEW REPORT ACTION APPEARED FROM ANOTHER USER AND REPORT IS IN VIEW
+                // We use the scroll position to determine whether the report should be marked as read and the new line indicator reset.
+                // If the user is scrolled up and no new line marker is set we will set it otherwise we will do nothing so the new marker
+                // stays in it's previous position.
+                if (this.currentScrollOffset === 0) {
+                    Report.readNewestAction(this.props.report.reportID);
+                    this.setState({newMarkerSequenceNumber: 0});
+                } else if (this.state.newMarkerSequenceNumber === 0) {
+                    this.setState({newMarkerSequenceNumber: currentLastSequenceNumber});
+                }
             }
         }
 
-        // Update the new marker position and last read action when we are closing the sidebar or moving from a small to large screen size
-        if (isReportFullyVisible && reportBecomeVisible) {
-            this.updateNewMarkerPosition(this.props.report.unreadActionCount);
+        // REPORT PREVIOUSLY HIDDEN BY SIDEBAR/LHN OR VIEW EXPANDED FROM MOBILE TO DESKTOP LAYOUT
+        // We update the new marker position, mark the report as read, and fetch new report actions
+        const sidebarClosed = prevProps.isDrawerOpen && !this.props.isDrawerOpen;
+        const reportBecameVisible = sidebarClosed || screenSizeIncreased;
+        if (isReportFullyVisible && reportBecameVisible) {
+            this.setState({
+                newMarkerSequenceNumber: this.props.report.unreadActionCount === 0
+                    ? 0
+                    : this.props.report.lastReadSequenceNumber + 1,
+            });
             Report.openReport(this.props.report.reportID);
         }
+
+        // REPORT PREVIOUSLY OPEN BUT USER NAVIGATED TO SIDEBAR/LHN
+        // When the user swipes back on the report or navigates to the LHN the ReportActionsView doesn’t unmount and just remains hidden.
```

Comment:
How's "swipes back on the report" different from "navigates to the LHN"?
```suggestion
        // When the user swipes back on the report or navigates to the LHN the ReportActionsView doesn't unmount and just remains hidden.
```

Reply:
I'll be more general here. I meant swipe or tap the back button. Both would be ways of navigating to the LHN.